libtransmission/transmission.h

```diff
@@ -24,15 +24,16 @@
 
 #include "tr-macros.h"
 
-using tr_file_index_t = uint32_t;
+using tr_file_index_t = size_t;
 using tr_piece_index_t = uint32_t;
 /* Assuming a 16 KiB block (tr_block_info::BlockSize), a 32-bit block index gives us a maximum torrent size of 64 TiB.
  * When we ever need to grow past that, change tr_block_index_t and  tr_piece_index_t to uint64_t. */
 using tr_block_index_t = uint32_t;
+using tr_byte_index_t = uint64_t;
 using tr_tracker_tier_t = uint32_t;
 using tr_tracker_id_t = uint32_t;
-using tr_byte_index_t = uint64_t;
 using tr_torrent_id_t = int;
+using tr_speed_t = size_t;
```

Comment:
I don't like the ambiguity of this `tr_speed_t` type. It implies some kind of structure but doesn't really deliver on that -- e.g. in lines 505 and 506 below, it alternately means `bytes_per_second` and `kilo_per_second`. 

Ultimately I'd like to use https://github.com/nholthaus/units for the job but that's out-of-scope for this PR.

What would you think about splitting `tr_speed_t` into `tr_bytes_per_second_t` and `tr_kilobytes_per_second_t`? Both could be `size_t` for now, and that `size_t` type could be replaced with a nholthaus/units type later.

Reply:
What about `tr_bps_t` and `tr_kbps_t`? (with a comment above the definition to give the verbose meaning)

Reply:
the trouble there is `bps` and `kbps` usually refers to bits and kilobits :dizzy:

Reply:
OK, let me try to double compile it.